transformers/modeling_bert.py

```diff
@@ -660,6 +660,13 @@ def forward(self, input_ids, attention_mask=None, token_type_ids=None, position_
         extended_attention_mask = extended_attention_mask.to(dtype=next(self.parameters()).dtype)  # fp16 compatibility
         extended_attention_mask = (1.0 - extended_attention_mask) * -10000.0
 
+        # If a 2D encoder attention mask is provided for the cross-attention
+        # we need to make broadcastabe to [batch_size, num_heads, seq_length, seq_length]
+        if encoder_attention_mask is not None:
+            encoder_attention_mask = encoder_attention_mask[:, None, None, :]
```

Comment:
Here we should be more open and accept `encoder_attention_mask` with a dimension of 2 or 3 (adding the relevant number of dimensions) like for the `attention_mask`.